pandas/io/orc.py

```diff
@@ -0,0 +1,108 @@
+""" orc compat """
+
+import distutils
+from typing import List, Optional
+
+from pandas.compat._optional import import_optional_dependency
+
+from pandas import DataFrame, get_option
+from pandas._typing import FilePathOrBuffer
+
+from pandas.io.common import get_filepath_or_buffer
+
+
+def get_engine(engine: str) -> "PyArrowImpl":
+    """ return our implementation """
+
+    if engine == "auto":
+        engine = get_option("io.orc.engine")
+
+    if engine == "auto":
+        # try engines in this order
+        try:
+            return PyArrowImpl()
+        except ImportError:
+            pass
+
+        raise ImportError(
+            "Unable to find a usable engine; "
+            "tried using: 'pyarrow'.\n"
+            "pyarrow is required for orc "
+            "support"
+        )
+
+    if engine not in ["pyarrow"]:
+        raise ValueError("engine must be 'pyarrow'")
+
+    return PyArrowImpl()
+
+
+class PyArrowImpl:
+    def __init__(self):
+        pyarrow = import_optional_dependency(
+            "pyarrow", extra="pyarrow is required for orc support."
+        )
+
+        # we require a newer version of pyarrow thaN we support for parquet
+        import pyarrow
+
+        if distutils.version.LooseVersion(pyarrow.__version__) < "0.13.0":
+            raise ImportError("pyarrow must be >= 0.13.0 for read_orc")
+
+        import pyarrow.orc
+
+        self.api = pyarrow
+
+    def read(
+        self, path: FilePathOrBuffer, columns: Optional[List[str]] = None, **kwargs
+    ) -> DataFrame:
+        path, _, _, _ = get_filepath_or_buffer(path)
+
+        py_file = self.api.input_stream(path)
+        orc_file = self.api.orc.ORCFile(py_file)
```

Comment:
Dug into this a little bit more and we should be able to just remove the `input_stream` call here and feed the `path` object which is a filepath or buffer directly into the ORCFile call. This will net us 0.9 compatibility in the process.

Reply:
great!

feel free to push a commit - or i will try this week

Reply:
Will try to find some time to push changes tomorrow 😄